ivy/__init__.py

```diff
@@ -224,6 +249,9 @@ class Node(str):
     float16,
     float32,
     float64,
+    complex64,
```

Comment:
I think we shouldn't have complex dtypes in the numeric category as most of the tests which are generating data with numeric datatypes could fail as they may not support complex dtypes. The same applies to all other files changed.